docs/blog/2025-09-16-creating-an-nx-plugin-for-biome.md

```diff
@@ -0,0 +1,448 @@
+
+We're working with a workspace that looks like this: a React application that depends on a few libraries. This workspace is small, but the process we're walking through will scale to any size workspace.
+
+![A workspace graph showing an application depending on three libraries.](/blog/images/2025-09-16/graph.avif)
+
+## Start with What You Know: Using Biome Without Nx
+
+Before thinking about Nx at all, let's see how far we can get knowing a little bit about Biome and a lot about the Node ecosystem.
+
+Looking at the [Biome getting started docs](https://biomejs.dev/guides/getting-started/), we need to:
+
+1. **Install the package**:
+
+```shell
+npm install --save-dev @biomejs/biome
+```
+
+2. **Create the root configuration** using their init command:
+
+```shell
+npx @biomejs/biome init
+```
+
+This creates a `biome.json` file at the root of our repo.
+
+3. **Run the linter**:
+
+```shell
+npx @biomejs/biome lint
+```
+
+Without thinking about Nx at all, we've successfully installed Biome and run a lint across our entire workspace. That's a good starting point, but in a monorepo, **we don't want to lint all files all the time**. We want to lint individual projects.
+
+## Using npm scripts
+
+Normally, to avoid writing long commands every time, we'd reach for npm scripts. Let's add this to our root `package.json`:
+
+```json {% fileName="package.json" %}
+{
+  "scripts": {
+    "biome-lint": "biome lint"
+  }
+}
+```
+
+Now we can run `npm run biome-lint` instead. But we're still linting the entire workspace.
+
+### Targeting Individual Projects
+
+Since our workspace uses npm workspaces, every project has its own `package.json`. We can add the same script to individual project package files:
+
+```json {% fileName="apps/biome-example/package.json" %}
+{
+  "scripts": {
+    "biome-lint": "biome lint"
+  }
+}
+```
+
+Now if we move into that directory and run `npm run biome-lint`, we're only linting that specific directory. When you run `biome lint` with no other options, it lints that directory and directories below it.
+
+## The Nx Integration you didn't realize was there
+
+Here's something you might not realize: **we've already started an Nx integration**. If you open Nx Console, you'll see that `biome-lint` is available as a target under the "npm scripts" area. Nx automatically incorporates npm scripts into available tasks.
+
+![Screenshot of Nx Console showing npm scripts available as targets](/blog/images/2025-09-16/npm-scripts.avif)
+
+So while `npm run biome-lint` works, we can also run this through Nx:
+
+```shell
+npx nx biome-lint biome-example
+```
+
+This gives us the exact same result. We can copy this script to other packages, like our navigation library, and suddenly we can run `biome-lint` on multiple projects individually.
```

Comment:
I would include a shell command that shows the run_many command with the -t biome-lint.